server/src/main/scala/ch.epfl.scala.index.server/routes/ProjectPages.scala

```diff
@@ -293,15 +314,12 @@ class ProjectPages(
                 pathEnd(
                   editForm(
                     form =>
-                      onSuccess(
-                        dataRepository.updateProject(
-                          Project.Reference(organization, repository),
-                          form
-                        )
-                      ) { _ =>
-                        Thread.sleep(1000) // oh yeah
-                        redirect(Uri(s"/$organization/$repository"), SeeOther)
-                    }
+                        onSuccess {
+                          updateProject(Project.Reference(organization, repository), form)
+                        } { _ =>
+                          Thread.sleep(1000) // oh yeah
```

Comment:
I know it was already there, but I don't understand why it's necessary, since onSuccess will wait to resolve the future first.

Reply:
I am not sure either. I think elasticsearch update are not striclty atomic. We give it some time to properly index the project, so that when the user is redirected the changes are likely to be effective.